plugins/lsp/schemastore.nix

```diff
@@ -0,0 +1,140 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+  helpers.vim-plugin.mkVimPlugin config {
+    name = "schemastore";
+    originalName = "SchemaStore.nvim";
+    defaultPackage = pkgs.vimPlugins.SchemaStore-nvim;
+
+    maintainers = [maintainers.GaetanLepage];
+
+    extraOptions = let
+      schemaOpts = let
+        schemaEntry = types.submodule {
+          options = {
+            name = mkOption {
+              type = types.str;
+              description = "The name of the schema.";
+            };
+
+            description = helpers.mkNullOrStr "A description for this schema.";
+
+            fileMatch = helpers.mkNullOrOption (with types; either str (listOf str)) ''
+              Which filename to match against for this schema.
+            '';
+
+            url = mkOption {
+              type = types.str;
+              description = "The URL of this schema.";
+            };
+          };
+        };
+      in {
+        select = helpers.defaultNullOpts.mkListOf types.str "[]" ''
+          A list of strings representing the names of schemas to select.
+          If this option is not present, all schemas are returned.
+          If it is present, only the selected schemas are returned.
+          `select` and `ignore` are mutually exclusive.
```

Comment:
Could link to [the catalogue](https://github.com/SchemaStore/schemastore/blob/master/src/api/json/catalog.json) for reference